### ISSUE TITLE
-c does not perform all compile time checks

### ISSUE THREAD

**Issue by coke** [rakudo/rakudo#2549]
## The Problem

Given the program
```
sub foo( Int $a, Int $b ) {...}
foo(1)
```

the ```-c``` compiler switch does not catch the compiler errors that running the program does.

(found in the doc site's FAQ)

## Expected Behavior

```
$ perl6 -c foo.p6
===SORRY!=== Error while compiling foo.p6
Calling foo(Int) will never work with declared signature (Int $a, Int $b)
at foo.p6:2
------> <BOL>⏏foo(1)     # ===SORRY!=== Error while co
```

## Actual Behavior

```
$ perl6 -c foo.p6
Syntax OK
$ perl6 foo.p6
===SORRY!=== Error while compiling foo.p6
Calling foo(Int) will never work with declared signature (Int $a, Int $b)
at foo.p6:2
------> <BOL>⏏foo(1)     # ===SORRY!=== Error while co
```

## Environment
* Operating system: 
```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G3025
```

* Compiler version (`perl6 -v`): 

```
$ perl6 -v
This is Rakudo version 2018.11-62-g01c56b484 built on MoarVM version 2018.11-33-gee6ea722c
implementing Perl 6.d.
```

**Comment by lizmat** [comment]
Yeah, the static optimizer has started to do much more.  And that is not called with `-c`.  Perhaps the easiest fix would be that it would.

**Comment by jnthn** [comment]
Yeah, it's just moving the syntax check stage after "the stage formerly known as optimize". :-) There's just a list of them somewhere, probably in `Perl6::Compiler`, so it's probably just moving the syntax check option to be further in the stage list.

**Comment by lizmat** [comment]
Hmmm... simply adding:

    self.optimize($past,|%adverbs);

just before the `say("Syntax OK")` does not do the right thing:

    Cannot find method 'ann' on object of type Perl6::Grammar
       at gen/moar/Perl6-Optimizer.nqp:52      
     from gen/moar/Perl6-Optimizer.nqp:47  
     from gen/moar/Perl6-Optimizer.nqp:901  
     from src/Perl6/Compiler.nqp:53
    ...

:-(
